components/backup-stream/src/utils.rs

```diff
@@ -627,6 +632,81 @@ impl AsyncRead for FilesReader {
     }
 }
 
+#[async_trait::async_trait]
+pub trait CompressionWriter: Sync + Send + 'static {
+    async fn write_all(&mut self, src: &[u8]) -> Result<()>;
+
+    async fn shutdown(&mut self) -> Result<()>;
```

Comment:
What does `shutdown` do?

Reply:
flush the data in memory into the file.
`shutdown` means close the write stream.

Reply:
Please add comments to these two methods.

Reply:
done